SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -163,3 +177,136 @@ bool UReferenceCountedLockingPolicy::ReleaseLockFromDelegate(AActor* ActorToRele
 	bool ReleaseSucceeded = ReleaseLock(LockToken);
 	return ReleaseSucceeded;
 }
+
+void UReferenceCountedLockingPolicy::OnOwnerUpdated(AActor* ActorChangingOwner)
```

Comment:
Why is this part of the reference counted policy and not another policy that can sit on top of it?

Reply:
Specifically as this is quite broad. This doesn't do a check to make sure you are in a hypothetical working set. It just doesn't it for all owned things.

Reply:
(Also we can pull out common functionality instead and encapsulate some things)

Reply:
This also looks like it might be very expensive.

Reply:
> Why is this part of the reference-counted policy and not another policy that can sit on top of it?

I'm happy to split this out from the reference-counted policy, maybe have another policy e.g. OwnershipActorSetsLockingPolicy? 

> (Also we can pull out common functionality instead and encapsulate some things)

Have refactored the loop at the bottom that's used in other places to a separate method

> This also looks like it might be very expensive.

It's quite expensive, however I assume the frequency of us trying to updating a locked Actor's hierarchy information should be quite infrequent.

Reply:
OwnershipHierarchyLockingPolicy maybe? I'd prefer to not call them actor sets until we know what an actor set is.

>It's quite expensive, however I assume the frequency of us trying to updating a locked Actor's hierarchy information should be quite infrequent.

Probably but we have to walk the list every time we lock anything. And will also walk that list a lot when things are locked.